zio-http/src/main/scala/zhttp/http/middleware/HttpMiddleware.scala

```diff
@@ -52,6 +52,17 @@ object HttpMiddleware {
 
   type RequestP[+A] = (Method, URL, List[Header]) => A
 
+  def ifThenElse[R, E](cond: RequestP[Boolean])(left: HttpApp[R, E], right: HttpApp[R, E]): HttpMiddleware[R, E] =
+    HttpMiddleware.fromHttpFunction((m, u, h) => if (cond(m, u, h)) left else right)
+
+  def ifThenElseM[R, E](cond: RequestP[UIO[Boolean]])(left: HttpApp[R, E], right: HttpApp[R, E]): HttpMiddleware[R, E] =
```

Comment:
```suggestion
  def ifThenElseM[R, E](cond: RequestP[ZIO[R, E, Boolean]])(left: HttpApp[R, E], right: HttpApp[R, E]): HttpMiddleware[R, E] =
```

Reply:
can we go with `cond: RequestP[ZIO[R, Option[E], Boolean]]` we are using `Option[E]` every were, like 
`def modifyM[R1 <: R, E1 >: E](f: RequestP[ZIO[R1, Option[E1], HttpMiddleware[R1, E1]]]): HttpMiddleware[R1, E1]`